### PR TITLE
Add --full-app flag to push command

### PR DESCRIPTION
**Pull request by spalladino**
Migrates a lightweight app to a full app.
Related to #165 (pending making lightweight app the default).

### CODE REVIEW

File: packages/cli/src/models/network/NetworkAppController.js [hunk 1]
```diff
@@ -9,8 +9,14 @@ import { AppProjectDeployer, SimpleProjectDeployer } from './ProjectDeployer';
 const log = new Logger('NetworkAppController');
 
 export default class NetworkAppController extends NetworkBaseController {
+  get isLightweight() {
+    return this.packageFile.isLightweight && !this.appAddress;
```
Comment (facuspagnuolo) [inline packages/cli/src/models/network/NetworkAppController.js]: I think we can rely only on the app address

File: packages/lib/src/project/SimpleProject.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ export default class SimpleProject  {
     delete this.implementations[contractName]
   }
 
-  registerImplementation(contractName, { address, bytecodeHash }) {
+  async registerImplementation(contractName, { address, bytecodeHash }) {
```
Comment (facuspagnuolo) [inline packages/lib/src/project/SimpleProject.js]: is it `async`?

File: packages/cli/src/commands/push.js [hunk 3]
```diff
@@ -16,13 +16,14 @@ const register = program => program
   .option('-d, --deploy-libs', 'deploys libraries to the network if there is no existing deployment')
   .option('--reset', 'redeploys all contracts (not only the ones that changed)')
   .option('-f, --force', 'ignores validation errors and deploys contracts')
+  .option('--full-app', 'deploys an App contract to manage the project')
```
Comment (facuspagnuolo) [inline packages/cli/src/commands/push.js]: leaving this comment to remind us that we are thinking of moving this flag to a new `publish` command

File: packages/lib/src/project/SimpleProject.js (hunk 2 above)
Comment (spalladino) [reply packages/lib/src/project/SimpleProject.js]: It's just to be consistent with the equivalent method in AppProject

File: packages/cli/src/models/network/NetworkAppController.js (hunk 1 above)
Comment (spalladino) [reply packages/cli/src/models/network/NetworkAppController.js]: isLightweight will return true by default soon!

**Comment by spalladino** [comment]
Closing as was already merged as part of https://github.com/zeppelinos/zos/pull/231

**Pull request closed without merging**